functable.c

```diff
@@ -54,6 +54,9 @@ void slide_hash_power8(deflate_state *s);
 #ifdef X86_AVX2
 void slide_hash_avx2(deflate_state *s);
 #endif
+#ifdef X86_AVX512
```

Comment:
Should be `#ifdef X86_AVX512_SLIDEHASH`

Reply:
If I was able to `undef X86_AVX512_SLIDEHASH` I could easily turn off this feature.

Reply:
The AVX2 one is wrong too and we are missing that def.

Reply:
I think we are missing some for AVX2.